cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -134,15 +129,14 @@
     this.dataTracerFactory = dataTracerFactory;
     this.namespaceAdmin = namespaceAdmin;
     this.metricsCollectionService = metricsCollectionService;
-    this.metricsQueryHelper = metricsQueryHelper;
     this.programNotificationSubscriberService = programNotificationSubscriberService;
     this.levelDBTableService = levelDBTableService;
     this.structuredTableAdmin = structuredTableAdmin;
     this.structuredTableRegistry = structuredTableRegistry;
   }
 
   @Override
-  public Future<PreviewRequest> startPreview(PreviewRequest previewRequest) throws Exception {
+  public Map.Entry<Future<PreviewRequest>, ProgramRunId> startPreview(PreviewRequest previewRequest) throws Exception {
```

Comment:
This change is required because PreviewRunner needs to know the program run id in order to stop it using ProgramLifeCycleService. Previously ProgramRunId was found in PreviewStore, however preview runner will not have access to the PreviewStore which is why we need to preserve it here.